scripts/sct_compute_mtr.py

```diff
@@ -73,8 +75,16 @@ def main(args=None):
     convert(fname_mt0, os.path.join(path_tmp, "mt0.nii"), dtype=np.float32)
     convert(fname_mt1, os.path.join(path_tmp, "mt1.nii"), dtype=np.float32)
 
-    # go to tmp folder
```

Comment:
actually, you can remove the whole part where we copy files to temp folder. We can simply open the data in the RAM, do the computation of mtr and then save to the specified output.

Reply:
if we do this, should we also remove the 'r' argument [(argument shown here)](https://github.com/YangHee-Min/spinalcordtoolbox/blob/10da873575e77cd1af81520931e3769ccb57c5fa/scripts/sct_compute_mtr.py#L142-L147)? According to my understanding, this argument is used to decide whether or not to delete the temporary files and would therefore be unnecessary if the temp folder does not exist to begin with (since we are removing that process).

Reply:
Changes to temp folder steps were made in commit [15f6de8](https://github.com/YangHee-Min/spinalcordtoolbox/commit/15f6de8d9e4cf1505687ead6e3208490f62dbff4)

Reply:
When we open the computation, we copy mt0.nii and mt1.nii in the wanted file, but these files are not necessary to open the mtr file. Since these files are created (or overwritten), should we delete these files at the end of the script to save space?

Reply:
you don't need to copy the mt1/mt0 files. Just open them, compute mtr, and save nifti file, that's all

Reply:
Ok understood! Steps removed in commit [e380997](https://github.com/YangHee-Min/spinalcordtoolbox/commit/e380997f44c0fd31c193a6c92250f7ed528b8e07)

Reply:
Correct me if I am wrong, but since opening .nii.gz files creates the corresponding .nii file, wouldn't calling sct_compute_mtr multiple times cause the creation of multiple copies of the .nii file? (thus needing a way to delete the .nii file)

Reply:
To open an image use spinalcordtoolbox.image.Image, not convert.

Reply:
Ok makes sense! Changes applied in commit [6d30511](https://github.com/YangHee-Min/spinalcordtoolbox/commit/6d30511f684581e32fa533079316134638a7bc24)